src/routes/accountOpeningRequest.ts

```diff
@@ -0,0 +1,106 @@
+import type { Request, Response } from "express";
+import HttpStatusCodes from "http-status";
+import moment from "moment";
+
+import {
+  getPerson,
+  savePerson,
+  saveAccountOpeningRequestToPersonId,
+  getPersonIdByAccountOpeningRequest,
+} from "../db";
+import {
+  AccountOpeningRequestStatus,
+  PersonWebhookEvent,
+} from "../helpers/types";
+import { triggerWebhook } from "../helpers/webhooks";
+import generateID from "../helpers/id";
+import { createAccount } from "../routes/accounts";
+
+export const createAccountOpeningRequest = async (
+  req: Request,
+  res: Response
+) => {
+  const data = req.body;
+
+  const personId = data.customer_id;
+
+  const person = await getPerson(personId);
+
+  const accountOpeningRequest = {
+    customer_id: data.customer_id,
+    customer_type: data.customer_type,
+    product_name: data.product_name,
+    account_type: data.account_type,
+    account_bic: data.account_bic,
+    account_currency: data.account_currency,
+    account_purpose: data.account_purpose,
+    id: generateID(),
+    status: AccountOpeningRequestStatus.INITIATED,
+    account_id: null,
+    iban: null,
+    created_at: moment().format("YYYY-MM-DD"),
+    updated_at: moment().format("YYYY-MM-DD"),
+    rejection_reason: {
+      failed_validation: null,
+      details: null,
+    },
+  };
+
+  person.accountOpeningRequests = person.accountOpeningRequests || [];
+  person.accountOpeningRequests.push(accountOpeningRequest);
+
+  await savePerson(person);
```

Comment:
We can secure it from thread races using redlock. (I'll create a PR targeting this one)
We would eventually replace the deprecated createPerson flow with this one and we would have issues when e2e user setup uses this function.